### PR TITLE
Add iOS build to CI

### PR DESCRIPTION
**Pull request by juan-lunarg**
Fix iOS build as well as minor Android cleanup since the problems for both platforms are so similar

### CODE REVIEW

**Review by ncesario-lunarg — APPROVED** [review]
LGTM! Thanks again Juan.

**Comment by juan-lunarg** [comment]
FYI @richard-lunarg

File: CMakeLists.txt [hunk 1]
```diff
@@ -59,6 +59,22 @@ if ("${CMAKE_BUILD_TYPE}" STREQUAL "")
...
+# They both have their own packaging (APP/APK).
+# Which makes regular executables/testing problematic.
+#
+# Currently the only deliverables for these platforms are
+# libraries (either STATIC or SHARED).
+#
+# Furthermore testing is equally problematic.
+if (IOS OR ANDROID)
+    set(ENABLE_GLSLANG_BINARIES OFF)
+    set(SPIRV_SKIP_EXECUTABLES ON)
```
Comment (juan-lunarg) [inline CMakeLists.txt:72]: For extra context the ndk-builds currently in glslang and spirv-tools also don't create executables. For the aforementioned reasons.

Comment (juan-lunarg) [reply CMakeLists.txt:72]: So this should also speed up our Android / iOS CI in downstream repos since we can avoid building extra executables.

**Review by arcady-lunarg — APPROVED** [review]

**Pull request merged**